dsl/src/main/scala/asobu/dsl/ControllerMethodBuilder.scala

```diff
@@ -16,54 +17,53 @@ trait ControllerMethodBuilder {
+    def apply[RMT, ExtractedRepr <: HList, V <: HList](
+      extractor: RequestExtractor[ExtractedRepr],
+      directive: Directive[RMT]
+    )(
+      implicit
+      combine: CombineWithInputArgs[ExtractedRepr, V, RMT]
+    ): handle[RMT, ExtractedRepr, V] = new handle(extractor, directive)
   }
 
   /**
    *
    * @tparam RMT request message type
    */
-  class handle[RMT, ExtractedRepr <: HList, FullRepr <: HList, InputRepr <: HList, V <: HList, K <: HList, TempFull <: HList](extractor: Extractor[ExtractedRepr], directive: Directive[RMT])(implicit
-    lgen: LabelledGeneric.Aux[RMT, FullRepr],
-                                                                                                                                                                                              restOf:  RestOf.Aux[FullRepr, ExtractedRepr, InputRepr],
-                                                                                                                                                                                              keys:    Keys.Aux[InputRepr, K],
-                                                                                                                                                                                              values:  Values.Aux[InputRepr, V],
-                                                                                                                                                                                              zip:     ZipWithKeys.Aux[K, V, InputRepr],
-                                                                                                                                                                                              prepend: Prepend.Aux[InputRepr, ExtractedRepr, TempFull],
-                                                                                                                                                                                              align:   Align[TempFull, FullRepr]) extends ProductArgs {
+  class handle[RMT, ExtractedRepr <: HList, V <: HList](
+      extractor: RequestExtractor[ExtractedRepr],
+      directive: Directive[RMT]
+  )(implicit combine: CombineWithInputArgs[ExtractedRepr, V, RMT]) extends ProductArgs {
+    def applyProduct(vs: V): EssentialAction = Action.async { req ⇒
+      extractor.run(req).map { extracted ⇒
+        val msg: RMT = combine(extracted, vs)
+        directive(req.map(_ ⇒ msg))
+      }.fold(Future.successful, identity).flatMap(identity)
+    }
+  }
 
-    import scala.concurrent.ExecutionContext.Implicits.global
+}
 
-    private def combine(inputRepr: InputRepr, extractedRepr: ExtractedRepr): RMT =
-      lgen.from(align(inputRepr ++ extractedRepr))
+object ControllerMethodBuilder extends ControllerMethodBuilder {
+  trait CombineWithInputArgs[ExtractedRepr <: HList, InputArgs <: HList, Out] {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Type parameter 'ExtractedRepr' does not conform with naming standard.](https://www.codacy.com/app/iHeartRadio/asobu/file/2260975629/issues/source?bid=3037259&fileBranchId=3178877#l53)